File: app/src/main/java/com/jdappel/beerinvestigator/ui/viewmodel/impl/BeerModule.java
package com.jdappel.beerinvestigator.ui.viewmodel.impl;

import com.jdappel.beerinvestigator.rest.BreweryDBApi;
import com.jdappel.beerinvestigator.ui.viewmodel.BeerViewModel;

import dagger.Module;
import dagger.Provides;

@Module
public class BeerModule {

    @Provides
    BeerViewModel provideBeerViewModel(BreweryDBApi service) {
        return new BeerViewModelImpl(service);
    }
}

File: app/src/main/java/com/jdappel/beerinvestigator/ui/viewmodel/impl/BeerModule.kt
package com.jdappel.beerinvestigator.ui.viewmodel.impl

import com.jdappel.beerinvestigator.rest.BreweryDBApi
import com.jdappel.beerinvestigator.ui.viewmodel.BeerViewModel

import dagger.Module
import dagger.Provides

@Module
class BeerModule {
    @Provides
    fun provideBeerViewModel(service: BreweryDBApi): BeerViewModel {
        return BeerViewModelImpl(service)
    }
}